### DISCUSSION TITLE
value masked/unmasked validation

### DISCUSSION THREAD

**Discussion by mashpie** [dm4t2/vue-currency-input#273] [Ideas]
this is kind-of a follow up to #230 

I was trying to resolve issues with validating the value and expected it to be a number as emitted by the component. Coming from your example in https://codesandbox.io/s/using-vue-currency-input-with-quasar-veevalidate-gokw6?file=/src/QCurrencyInput.vue:379-581

The value in validation function contains the formatted string:

```js
const { errorMessage, value } = useField("amount", (value) => !!value); // ---> reads 1.234,00 €
```

while the bound value correctly contains the plain number:

```vue
    <QCurrencyInput
      v-model="value"
      :options="{
        currency: 'USD',
        distractionFree: false,
      }"
    />
    Number value: <code>{{ value }}</code>
    <!-- reads 1234 -->
```

Right now I don't see any api to get the plain unmasked value within that validation method, right? So I need to convert that string back to a number... but I hope this is not intended and I am simply missing something, do I?

**Comment by dm4t2** [comment]
Hi, thanks for the clarification. Now I understand the actual issue of #222 and #230 reported by @chaosKP.
Using `props.modelValue` in the validation function does not work, since it contains always the previously inputted number. 

I will provide an API for parsing the formatted value like in [v1](https://vue-currency-input-v1.netlify.app/api/#parse).

**Reply by mashpie** [reply]
Excellent! :)

**Reply by dm4t2** [reply]
I introduced a new API function `parse` in [v2.1.0](https://github.com/dm4t2/vue-currency-input/releases/tag/2.1.0) 🎉 

Docs: https://dm4t2.github.io/vue-currency-input/api.html#parse
Example: https://codesandbox.io/s/using-vue-currency-input-with-quasar-veevalidate-gokw6?file=/src/QCurrencyInput.vue
